data-products/src/main/scala/org/sunbird/analytics/exhaust/collection/BaseCollectionExhaustJob.scala

```diff
@@ -102,27 +101,51 @@ trait BaseCollectionExhaustJob extends BaseReportsJob with IJob with OnDemandExh
-    val result = for (request <- requests) yield {
-      if (validateRequest(request)) {
-        val res = CommonUtil.time(processRequest(request, custodianOrgId, userCachedDF))
-        JobLogger.log("The Request is processed", Some(Map("requestId" -> request.request_id, "timeTaken" -> res._1, "remainingRequest" -> totalRequests.getAndDecrement())), INFO)
-        res._2
-      } else {
-        JobLogger.log("Invalid Request", Some(Map("requestId" -> request.request_id, "remainingRequest" -> totalRequests.getAndDecrement())), INFO)
-        markRequestAsFailed(request, "Invalid request")
+    JobLogger.log(s"Total distinct requests are: ${totalRequests.getAndDecrement()}", None, INFO)
+    val result: Array[JobRequest] = for (request <- requests) yield {
+      try {
+        if (validateRequest(request)) {
+          updateRequests(Array(request)) // Set the request status to PROCESSING when it picked up for processing
+          val res = CommonUtil.time(processRequest(request, custodianOrgId, userCachedDF))
+          JobLogger.log("The Request is processed", Some(Map("requestId" -> request.request_id, "timeTaken" -> res._1, "requestStatus" -> res._2.status, "remainingRequest" -> totalRequests.getAndDecrement())), INFO)
+          saveRequests(storageConfig, Array(res._2))
+          res._2
+        } else {
+          JobLogger.log("Invalid Request", Some(Map("requestId" -> request.request_id, "requestStatus" -> "FAILED", "remainingRequest" -> totalRequests.getAndDecrement())), INFO)
+          val failedRequest = markRequestAsFailed(request, "Invalid Request")
+          updateRequests(Array(failedRequest))
+          failedRequest
+        }
+      }
+      catch {
+        case ex: Exception => {
+          JobLogger.log(s"The Request is failed to process due to ${ex.getMessage}", Some(Map("requestId" -> request.request_id, "requestStatus" -> "FAILED",  "remainingRequest" -> totalRequests.getAndDecrement())), INFO)
+          ex.printStackTrace()
+          val failedRequest = markRequestAsFailed(request, ex.getMessage)
+          updateRequests(Array(failedRequest))
+          failedRequest
+        }
       }
     }
-    saveRequests(storageConfig, result);
-    Metrics(totalRequests = Some(requests.length), failedRequests = Some(result.count(x => x.status.toUpperCase() == "FAILED")), successRequests = Some(result.count(x => x.status.toUpperCase == "SUCCESS")))
+    Metrics(totalRequests = Some(requests.length), failedRequests = Some(result.count(x => x.status.toUpperCase() == "FAILED")), successRequests = Some(result.count(x => x.status.toUpperCase == "SUCCESS")), Some(filteredRequests.getOrElse("duplicate", new Array[JobRequest](0)).length))
+  }
+
+  def filterRequests(jobRequests: DataFrame): Map[String, Array[JobRequest]] = {
+    val encoder = Encoders.product[JobRequest]
+    val uniqueRequestsDf = jobRequests.dropDuplicates("tag")
```

Comment:
Shouldn't the requests be deduped on tag and jobid? This would drop the same tag requests of other job

Reply:
`jobid` is not required, since we are fetching the requests by `JobId` only on top of that result we are doing de-dup 
`filterRequests(getRequests(jobId()))`